thumb.php

```diff
@@ -26,24 +26,27 @@
     Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
-// [catthumb id="Category ID" link="ture"]
+// [catthumb id="Category ID" link="true"]
 
 function dirtysuds_category_thumb( $atts ) {
 	extract( shortcode_atts( array(
 		'id' => 1,
 		'link' => 'true',
+		'post_type' => 'post',
+		'size' => 'thumbnail'
 	), $atts ) );
-
 	$embed = get_transient( 'dirtysuds_category_thumb' . $id . $link );
 	if( $embed ) return $embed;
 
 	$embed = '';
-	$posts = get_posts('cat='.$id.'&showposts=1&meta_key=_thumbnail_id');
-	if( $posts[0] ) {
-		$embed = get_the_post_thumbnail($posts[0]->ID,'thumbnail');
+	$posts = get_posts('cat='.$id.'&post_type='.$post_type.'&showposts=1&meta_key=_thumbnail_id');
+	if( isset( $posts[0] ) ) {
+		$embed = get_the_post_thumbnail($posts[0]->ID, $size);
 		if ($link == 'true') {
 			$embed = '<a href="'.get_category_link($id).'" title="'.get_cat_name($id).'">' . $embed . '</a>';
 		}
+	} else {
+		return '';
```

Comment:
This seems unnecessary.

`$embed` is already set to an empty string on line 40/41. If the `if` statement is false, it will already return an empty string without this `else`.